frontend/deb/debroot.go

```diff
@@ -0,0 +1,226 @@
+}
+
+// Debroot creates a debian root directory suitable for use with debbuild.
+// This does not include sources in case you want to mount sources (instead of copying them) later.
+func Debroot(sOpt dalec.SourceOpts, spec *dalec.Spec, in llb.State, target, dir string, opts ...llb.ConstraintsOpt) (llb.State, error) {
+	control, err := controlFile(spec, in, target, dir)
+	if err != nil {
+		return llb.Scratch(), errors.Wrap(err, "error generating control file")
+	}
+
+	rules, err := Rules(spec, in, dir)
+	if err != nil {
+		return llb.Scratch(), errors.Wrap(err, "error generating rules file")
+	}
+
+	changelog, err := Changelog(spec, in, target, dir)
+	if err != nil {
+		return llb.Scratch(), errors.Wrap(err, "error generating changelog file")
+	}
+
+	if dir == "" {
+		dir = "debian"
+	}
+
+	base := llb.Scratch().File(llb.Mkdir(dir, 0o755), opts...)
+	installers := createInstallScripts(spec, dir)
+
+	debian := base.
+		File(llb.Mkfile(filepath.Join(dir, "compat"), 0o440, []byte("10")), opts...).
+		File(llb.Mkdir(filepath.Join(dir, "source"), 0o755), opts...).
+		File(llb.Mkfile(filepath.Join(dir, "source/format"), 0o440, []byte("3.0 (quilt)")), opts...).
+		File(llb.Mkfile(filepath.Join(dir, "source/options"), 0o440, []byte("create-empty-orig")), opts...).
+		File(llb.Mkdir(filepath.Join(dir, "dalec"), 0o755), opts...).
+		File(llb.Mkfile(filepath.Join(dir, "source/include-binaries"), 0o440, append([]byte("dalec"), '\n')), opts...)
+
+	states := []llb.State{control, rules, changelog, debian}
+	states = append(states, installers...)
+
+	dalecDir := base.
+		File(llb.Mkdir(filepath.Join(dir, "dalec"), 0o755), opts...)
+
+	states = append(states, dalecDir.File(llb.Mkfile(filepath.Join(dir, "dalec/build.sh"), 0o700, createBuildScript(spec)), opts...))
+	states = append(states, dalecDir.File(llb.Mkfile(filepath.Join(dir, "dalec/patch.sh"), 0o700, createPatchScript(spec)), opts...))
+	states = append(states, dalecDir.File(llb.Mkfile(filepath.Join(dir, "dalec/fix_file_backed_sources.sh"), 0o700, fixupFileBackedSourcesScript(spec)), opts...))
+
+	patchDir := dalecDir.File(llb.Mkdir(filepath.Join(dir, "dalec/patches"), 0o755), opts...)
+	sorted := dalec.SortMapKeys(spec.Patches)
+	for _, name := range sorted {
+		pls, err := sourcePatchesDir(sOpt, patchDir, filepath.Join(dir, "dalec/patches"), name, spec, opts...)
+		if err != nil {
+			return llb.Scratch(), errors.Wrap(err, "error creating patch directory")
+		}
+		states = append(states, pls...)
+	}
+
+	return dalec.MergeAtPath(in, states, "/"), nil
+}
+
+func fixupFileBackedSourcesScript(spec *dalec.Spec) []byte {
+	// first, don't bother doing this for patches, which we don't include in our source tarballs.
+	patches := map[string]struct{}{}
+	for _, patchList := range spec.Patches {
+		for _, patch := range patchList {
+			patches[patch.Source] = struct{}{}
+		}
+	}
+
+	buf := bytes.NewBuffer(nil)
+	buf.WriteString("#!/usr/bin/env sh\n\n")
+	buf.WriteString("set -ex\n\n")
```

Comment:
nit: can the headers maybe of the scrips be consistent between fixupBackedSourceScript, createBuildScript and createPatchScript? It seems it have a space on it?
```go
	fmt.Fprintln(buf, "#!/usr/bin/env sh")
	fmt.Fprintln(buf)
```

Reply:
Done.